### PR TITLE
Fix #198

### LINKED ISSUE #198: Responding with HTTP 204
I am having trouble understanding how to respond with HTTP 204 (no content) using `gin.Context`.

Tried `context.JSON(204, nil)` or just `gin.H{}` but both complain that the body is not empty.

```
Error #01: http: request method or response status code does not allow body
     Meta: [<nil>]
```

Any help or pointer much appreciated!

Thanks

### PR DESCRIPTION
**Pull request by javierprovecho**
@appleboy @tboerger this is a solution for #198.

I'm intrigued why [`http.ResponseWriter`](https://golang.org/src/net/http/server.go?h=ResponseWriter#L86) is not returning [`http.ErrBodyNotAllowed`](https://golang.org/src/net/http/server.go?h=ErrBodyNotAllowed#L39).

I don't have time to look into it right now, hope you can, or we can review it later this weekend.

There is no breaking changes except to those who import directly the following:
- [github.com/gin-gonic/gin/render](https://github.com/gin-gonic/gin/tree/master/render)
- [github.com/manucorportat/sse](https://github.com/manucorporat/sse)

Maybe I'm leaving something behind, so more eyes on this are welcome.

### CODE REVIEW

**Comment by javierprovecho** [comment]
Some tests are failing because `\n` at the end of body.

File: sse/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Server-sent events (SSE) is a technology where a browser receives automatic updates from a server via HTTP connection. The Server-Sent Events EventSource API is [standardized as part of HTML5[1] by the W3C](http://www.w3.org/TR/2009/WD-eventsource-20091029/).
+
+- [Real world demostration using Gin](http://sse.getgin.io/)
+- [Read this great SSE introduction by the HTML5Rocks guys](http://www.html5rocks.com/en/tutorials/eventsource/basics/)
+- [Browser support](http://caniuse.com/#feat=eventsource)
+
+##Sample code
+
+```go
+import "github.com/manucorporat/sse"
```
Comment (appleboy) [inline sse/README.md]: Wrong import path?

File: sse/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#Server-Sent Events [![GoDoc](https://godoc.org/github.com/manucorporat/sse?status.svg)](https://godoc.org/github.com/manucorporat/sse) [![Build Status](https://travis-ci.org/manucorporat/sse.svg)](https://travis-ci.org/manucorporat/sse)
+
+Server-sent events (SSE) is a technology where a browser receives automatic updates from a server via HTTP connection. The Server-Sent Events EventSource API is [standardized as part of HTML5[1] by the W3C](http://www.w3.org/TR/2009/WD-eventsource-20091029/).
+
+- [Real world demostration using Gin](http://sse.getgin.io/)
```
Comment (appleboy) [inline sse/README.md]: http://sse.getgin.io Internal link?

File: sse/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#Server-Sent Events [![GoDoc](https://godoc.org/github.com/manucorporat/sse?status.svg)](https://godoc.org/github.com/manucorporat/sse) [![Build Status](https://travis-ci.org/manucorporat/sse.svg)](https://travis-ci.org/manucorporat/sse)
```
Comment (appleboy) [inline sse/README.md]: Wrong travis url. We need to remove link.

File: sse/README.md (hunk 1 above)
Comment (javierprovecho) [inline sse/README.md]: https://github.com/gin-contrib/sse

File: sse/README.md (hunk 2 above)
Comment (javierprovecho) [inline sse/README.md]: no, it was a demo for hacker news day, i removed it https://github.com/gin-contrib/sse

File: sse/README.md (hunk 3 above)
Comment (javierprovecho) [inline sse/README.md]: fixed in https://github.com/gin-contrib/sse

**Comment by javierprovecho** [comment]
@appleboy moved sse to https://github.com/gin-contrib/sse

**Comment by javierprovecho** [comment]
@appleboy pr updated. Later when this is merged, and a new version is released, we should move the /render and /binding imports to gopkg.

**Comment by appleboy** [comment]
@javierprovecho Good Jobs. LGTM

**Comment by appleboy** [comment]
@javierprovecho We will move `render` and `binding` to https://github.com/gin-contrib ?

**Comment by javierprovecho** [comment]
@appleboy maybe, i'm not sure. for now just to update the import statement from `github.com...` to `gopkg.in...` to really pull/use the same commit when releasing the next version (for being visible from `gin.v1` or master), if not, it will pull twice both from `gopkg.in` and `github.com`.

**Pull request merged**